src/lib/utils.ts

```diff
@@ -11,24 +11,31 @@ const debug = debugLib( '@automattic/vip:lib:utils' );
  *
  * @param {Function} fn       A function to poll
  * @param {number}   interval Poll interval in milliseconds
- * @param {Function} isDone   A function that accepts the return of `fn`. Stops the polling if it returns true
+ * @param {Function} isDone   A function that accepts the return of `fn`. Stops the polling if it returns true. If not provided, the polling stops when `fn` returns a truthy value.
+ * @param {number}   cutOff   Maximum time to poll in milliseconds
  * @return {Promise}          A promise which resolves when the polling is done
  * @throws {Error}            If the fn throws an error
  */
 export async function pollUntil< T >(
 	fn: () => Promise< T >,
 	interval: number,
-	isDone: ( v: T ) => boolean
+	isDone?: ( v: T ) => boolean,
+	cutOff?: number
```

Comment:
```suggestion
	cutOff: number
```

should this be optional? Also can we rename to `maxAllowedDuration` ?

Reply:
I think it is fine to keep it optional, as it is already being used in other places, and we haven't faced any deadlock on them yet.